examples/dataflow/chatbot/operations.py

```diff
@@ -0,0 +1,223 @@
+                continue
+            data = json.loads(data.strip())
+            message = data["text"]
+            # Only listen to messages directed to bot
+            if f"@{botname}" not in message:
+                continue
+            yield {"message": message}
+
+
+@op(
+    inputs={"message": TOSEND, "room_id": ROOMID},
+    config_cls=GitterChannelConfig,
+    imp_enter={
+        "config": lambda self: self.config.config,
+        "session": lambda self: ClientSession(trust_env=True),
+    },
+    ctx_enter={"cfg": (lambda self: self.parent.config())},
+)
+async def send_message(self, message, room_id):
+    access_token = await self.cfg.get("access_token")
+    headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "Authorization": f"Bearer {access_token}",
+    }
+    try:
+        message = json.loads(message)
+        message = json.dumps(message, indent=4, sort_keys=True)
+    except:
+        pass
+
+    # For new line we need \\n,else Gitter api
+    # responds with 'Bad Request'
+    message = message.replace("\n", "\\n")
+    # Replacing all double quotes with single quotes
+    # Because we are quoting 'message' in double quotes
+    message = message.replace('"', "'")
+    data = '{"text":' + f'"{message}"' + "}"
+
+    url = f"https://api.gitter.im/v1/rooms/{room_id}/chatMessages"
+    async with self.parent.session.post(
+        url, data=data, headers=headers
+    ) as resp:
+        response = await resp.json()
+        return
+
+
+@op(
+    inputs={"message": MESSAGE,},
+    outputs={"message": TOSEND},
+    config_cls=GitterChannelConfig,
+    imp_enter={"config": (lambda self: self.config.config)},
+    ctx_enter={"cfg": (lambda self: self.parent.config())},
+)
+async def interpret_message(self, message):
+    greet = ["hey", "hello", "hi"]
+    for x in greet:
+        if x in message.lower():
+            return {"message": "Hey Hooman ฅ^•ﻌ•^ฅ"}
+
+    def extract_data(raw_data):
+        raw_data = raw_data.split("data:")
+        data = {"model-data": raw_data[1]}
+        raw_data = raw_data[0].split("\n")
+        for x in raw_data:
+            k, *v = x.split(":")
+            if isinstance(v, list):  # for features
+                v = ":".join(v)
+            k = k.strip()
+            v = v.strip()
+            if k:  # avoid blank
+                data[k] = v
+        return data
+
+    # Removing username from message
+    # The regex matches @ followed by anything that
+    # is not a whitespace in the first group and
+    # the rest of the string in the second group.
+    # We replace the string by the second group.
+    message = re.sub(r"(@[^\s]+)(.*)", r"\2", message).strip()
+    access_token = await self.cfg.get("access_token")
+
+    if message.lower().startswith("train model"):
+        return {"message": "Gimme more details!!"}
+
+    elif message.lower().startswith("predict:"):
+        # only replace first occurence of predict
```

Comment:
Let's make sure that all comments start with a capitalized word

```suggestion
        # Only replace first occurrence of predict
```